### PR TITLE
Train one model per telescope type

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

File: ctapipe/ml/sklearn.py [hunk 1]
```diff
@@ -29,9 +29,16 @@ class Model(Component):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.model = SUPPORTED_MODELS[self.model_cls](**self.model_config)
+
+        # to verify settings
+        self.new_model()
```
Comment (nbiederbeck) [inline ctapipe/ml/sklearn.py:34]: what kind of exception would this raise? something that is easily understandable? or should we use a `try` block with some kind of `BrokenConfigException`?

Comment (maxnoe) [reply ctapipe/ml/sklearn.py:34]: Either a key error for an unsupported model or a sklearn error im case of unknown options, the ones I found were pretty clear

Comment (nbiederbeck) [reply ctapipe/ml/sklearn.py:34]: Great!

**Comment by nbiederbeck** [comment]
this now includes #1956 and #1957, is this intended?

File: ctapipe/ml/sklearn.py [hunk 2]
```diff
@@ -29,27 +29,43 @@ class Model(Component):
...
         self.unit = table[self.target].unit
-        self.model.fit(X, y)
+        self.models[key].fit(X, y)
 
-    def predict(self, table):
+    def predict(self, key, table):
+        if key not in self.models:
+            raise KeyError(
+                f"No model available for key {key},"
+                f" available modes: {self.models.keys()}"
```
Comment (nbiederbeck) [inline ctapipe/ml/sklearn.py:65]: typo: models

File: ctapipe/ml/sklearn.py [hunk 3]
```diff
@@ -127,9 +141,15 @@ class Classifier(Model):
 
     invalid_class = Integer(-1).tag(config=True)
 
-    def predict(self, table):
+    def predict(self, key, table):
+        if key not in self.models:
+            raise KeyError(
+                f"No model available for key {key},"
+                f" available modes: {self.models.keys()}"
```
Comment (nbiederbeck) [inline ctapipe/ml/sklearn.py:148]: typo: models

**Pull request merged**